File: src/main/scala/com/github/mauricio/postgresql/exceptions/UnsupportedAuthenticationMethodException.scala
package com.github.mauricio.postgresql.exceptions

/**
 * User: mauricio
 * Date: 3/30/13
 * Time: 11:23 PM
 */
class UnsupportedAuthenticationMethodException ( val authenticationType : Int )
  extends IllegalArgumentException ( "Unknown authentication method -> '%s'".format(authenticationType) )

File: db-async-common/src/main/java/com/github/jasync/sql/db/exceptions/UnsupportedAuthenticationMethodException.kt
package com.github.jasync.sql.db.exceptions

@Suppress("MemberVisibilityCanBePrivate", "CanBeParameter", "RedundantVisibilityModifier")
public class UnsupportedAuthenticationMethodException(val authenticationType: String) :
    DatabaseException("Unknown authentication method -> '$authenticationType'") {

    constructor(authType: Int) :
        this(authType.toString())
}
